optuna/integration/chainer.py

```diff
@@ -10,7 +10,7 @@
     from chainer.training.triggers import ManualScheduleTrigger
 
 if not _imports.is_successful():
-    Extension = object  # NOQA
+    Extension = object  # type: ignore # NOQA
```

Comment:
I'm not sure why we need `type: ignore` since the other files like `keras.py` or `allennlp.py` can use the same trick to assign dummy type. `mypy` happens to reach here and raise the following errors, but I think it should skip this line. Do you have any idea about it?

```console
optuna/integration/chainer.py:13: error: Cannot assign to a type
optuna/integration/chainer.py:13: error: Incompatible types in assignment (expression has type "Type[object]", variable has type "Type[Extension]")
```

Reply:
I believe there's some sort of static analysis. Not sure about the details but I'm also wondering if not the error is expected.

```python
int = object

from chainer.training.extension import Extension
Extension = object  # Error.

if False:
    Extension = object  # Error.

from multiprocessing import Event
Event = object  # Error.

from numpy import ndarray
ndarray = object

from tensorflow.keras.callbacks import Callback
Callback = object

from optuna import Study
Study = object  # Error.
```

Reply:
Thank you for the comments. I don't have any idea how to fix this issue. As I commented on the issue https://github.com/optuna/optuna/issues/1801#issuecomment-693798099, it will be future work after this PR.

Reply:
@hvy @HideakiImamura 

Thank you for your comments. In that case, I'm wondering why no errors occur in the case of `keras` or `allennlp`.
I also think we can work on it as a separate issue.